Userland/Utilities/man.cpp

```diff
@@ -91,13 +91,12 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
                 break;
             }
         }
-        if (!section) {
-            warnln("No man page for {}", name);
-            exit(1);
-        }
     }
-
     auto filename = make_path(section);
+    if (!(access(filename.characters(), R_OK) == 0)) {
+        warnln("No man page for {}", name);
+        exit(1);
+    }
```

Comment:
Why not just check `section == nullptr` here right before `filename = ...`? That would check for the right thing automatically.
I can't quite remember what AK/Format does to `(char*)nullptr`, but if it doesn't crash, you can get that access() to return 0 now:
```console
# mkdir_and_touch '/usr/share/man/man(null)/gibberish.md
$ man gibberish
```

Reply:
So something like this?
```
if (section == nullptr || !(access(filename.characters(), R_OK) == 0)) {
```

Reply:
I think I got something better now:
```cpp
auto filename = make_path(section);
if (section == nullptr) {
    warnln("No man page for {}", name);
    exit(1);
} else if (!(access(filename.characters(), R_OK) == 0)) {
    warnln("No man page for {} in section {}", name, section);
    exit(1);
}
```
This also reports back that the man page wasn't found in the specified section.